### PR TITLE
ROX-21099: validate CLI input

### PR DESCRIPTION
**Pull request by tommartensen**

### CODE REVIEW

File: cmd/infractl/cluster/create/create.bats [hunk 1]
```diff
@@ -141,6 +141,12 @@ setup() {
   assert_output --partial "Error: The name does not match the requirements."
 }
 
+@test "cannot create a cluster with an invalid lifespan" {
```
Comment (gavin-stackrox) [inline cmd/infractl/cluster/create/create.bats:144]: :bat:

File: cmd/infractl/cluster/utils/validate.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+		return err
+	}
+	if !match {
+		return errors.New("key is invalid format")
+	}
+
+	if value == "" {
+		return errors.New("value is empty")
+	}
+	match, err = regexp.MatchString(`^[a-zA-Z0-9:?/.-]+$`, value)
```
Comment (gavin-stackrox) [inline cmd/infractl/cluster/utils/validate.go:70]: Limiting values could be problematic. What about:
```suggestion
	match, err = regexp.MatchString(`^[[:ascii:]]+$`, value)
```

**Review by gavin-stackrox — COMMENTED** [review]
What about a length limit as mentioned in the JIRA?

**Comment by tommartensen** [comment]
> What about a length limit as mentioned in the JIRA?

I think that would only be reasonable for cluster names, where we have implemented it before: https://github.com/stackrox/infra/blob/master/cmd/infractl/cluster/create/command.go#L164

File: cmd/infractl/cluster/utils/validate.go (hunk 2 above)
Comment (tommartensen) [reply cmd/infractl/cluster/utils/validate.go:70]: Excluding certain characters from ASCII was intentional, like 
```
`;
```

I looked at the parameters, most of them are boolean, numbers (node counts), images and single word strings like `Passthrough`. I created the regex based on that.

**Review by gavin-stackrox — APPROVED** [review]

File: cmd/infractl/cluster/utils/validate.go (hunk 2 above)
Comment (gavin-stackrox) [reply cmd/infractl/cluster/utils/validate.go:70]: ok. lets see how it goes.

**Pull request merged**